neuralop/models/rno.py

```diff
@@ -0,0 +1,164 @@
+        self.residual = residual
+
+        # if lifting_channels is passed, make lifting an MLP with a hidden layer of size lifting_channels
+        if self.lifting_channels:
+            self.lifting = MLP(in_channels=in_channels, out_channels=self.hidden_channels, hidden_channels=self.lifting_channels, n_layers=2, n_dim=self.n_dims)
+        # otherwise, make it a linear layer
+        else:
+            self.lifting = MLP(in_channels=in_channels, out_channels=self.hidden_channels, hidden_channels=self.hidden_channels, n_layers=1, n_dim=self.n_dims)
+
+        module_list = [RNO_layer(n_modes, hidden_channels, return_sequences=True, output_scaling_factor=self.output_scaling_factor[i], fft_norm=fft_norm, factorization=factorization, separable=separable)
+                                     for i in range(n_layers - 1)]
+        module_list.append(RNO_layer(n_modes, hidden_channels, return_sequences=False, output_scaling_factor=self.output_scaling_factor[-1], fft_norm=fft_norm, factorization=factorization, separable=separable))
+        self.layers = nn.ModuleList(module_list)
+
+        # if projection_channels is passed, make lifting an MLP with a hidden layer of size lifting_channels
+        if self.projection_channels:
+            self.projection = MLP(in_channels=self.hidden_channels, out_channels=out_channels, hidden_channels=self.projection_channels, n_layers=2, n_dim=self.n_dims)
+        # otherwise, make it a linear layer
+        else:
+            self.projection = MLP(in_channels=self.hidden_channels, out_channels=out_channels, hidden_channels=out_channels, n_layers=1, n_dim=self.n_dims)
+    
+    def forward(self, x, init_hidden_states=None): # h must be padded if using padding
+        # x shape (batch, timesteps, dim, dom_size1, dom_size2, ..., dom_sizen)
+        batch_size, timesteps = x.shape[:2]
+        dim = x.shape[2]
+        dom_sizes = x.shape[3 : 3 + self.n_dims]
+        x_size = len(x.shape)
+
+        if init_hidden_states is None:
+            init_hidden_states = [None] * self.n_layers
+        
+        x = self.lifting(x.reshape(batch_size * timesteps, *x.shape[2:]))
+        x = x.reshape(batch_size, timesteps, *x.shape[1:])
+
+        if self.domain_padding:
+            x = self.domain_padding.pad(x)
+
+        final_hidden_states = []
+        for i in range(self.n_layers):
+            pred_x = self.layers[i](x, init_hidden_states[i])
+            if i < self.n_layers - 1:
+                if self.residual:
+                    x = x + pred_x
+                else:
+                    x = pred_x
+                final_hidden_states.append(x[:, -1])
+            else:
+                x = pred_x
+                final_hidden_states.append(x)
+        h = final_hidden_states[-1]
+
+        if self.domain_padding:
+            h = h.unsqueeze(1) # add dim for padding compatibility
+            h = self.domain_padding.unpad(h)
+            h = h[:,0] # remove extraneous dim
```

Comment:
Why not just squeeze?

Reply:
Are you referring to `squeeze` for the last line? In general, if I were to `squeeze` here, I would be wary of accidentally deleting the `out_channels` dimension if it is 1.

Reply:
Yes, just squeeze that particular dimension, ie. h = h.squeeze(1)